### PR TITLE
Fix some packages imports and fix windows hot reload problems

### LINKED ISSUE #88: "Hot reload failed" all the time after attempting to import http
**Describe the bug**
After attempting to import `package:http/http.dart`, for all inputs I get `Error: Hot reload failed, maybe because code has syntax error?`, even if the code is syntactically correct.

**To Reproduce**
1. Install the shell using `dart pub global activate interactive`
2. Open the shell using `interactive`
3. `!dart pub add http`
4. `import 'package:http/http.dart`
Any line from now on will error out with `Error: Hot reload failed, maybe because code has syntax error?`

**Expected behavior**
The code executes correctly, and the library is imported.

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: Arch Linux x86_64
 - Browser: Firefox 116.0.1

**Additional context**
The issue only occurs after attempting to import **ANY** package.

### PR DESCRIPTION
**Pull request by davidmartos96**
Fixes #88 

Hello! I stumbled across the issue when importing certain packages. It turns out that if we provide the path of `.dart_tool/package_config.json` to the workspace isolate, the interpreter can correctly handle those imports.

In order to generate the package_config.json we run `dart pub get` in the workspace without piping the output to the interpreter. 

I've also added a few E2E tests for package imports.

### CODE REVIEW

**Comment by fzyzcjy** [comment]
Great job! I will review after the CI is green (given that ubuntu and macos passes, it may not be hard to fix windows - since dart is cross-platform and this lib does not utilize special properties of a platform)

**Comment by davidmartos96** [comment]
> Great job! I will review after the CI is green (given that ubuntu and macos passes, it may not be hard to fix windows - since dart is cross-platform and this lib does not utilize special properties of a platform)

Great! 

I'm not entirely sure why it's failing on Windows when running "dart pub get" on the workspace. :thinking:

**Comment by davidmartos96** [comment]
@fzyzcjy I think I encountered the issue in this PR, which was how the process was being executed (no /bin/sh on Windows), but unfortunately there seems to be another issue not related with the new code.

I'm trying to run the master branch tests on Windows and they are failing with the same error as the one in this job. It's like the evaluate code isn't consistent. It may return null, or repeat some previous values. Maybe a Hot reload issue?

Would it be possible for you to re-run the Github actions on master? Just to see how CI runs master as of today.

File: packages/interactive/lib/src/workspace_isolate.dart [hunk 1]
```diff
@@ -70,5 +86,22 @@ Future<Isolate> _spawnUriWithErrorHandling(Uri uri) async {
     null,
     onError: errorPort.sendPort,
     onExit: exitPort.sendPort,
+    packageConfig: packageConfigUri,
+  );
+}
+
+Future<void> _runPubGet({required String workingDirectory}) async {
+  final process = await Process.start(
+    '/bin/sh',
```
Comment (JordyScript) [inline packages/interactive/lib/src/workspace_isolate.dart]: this line won't work on windows

Comment (davidmartos96) [reply packages/interactive/lib/src/workspace_isolate.dart]: @JordyScript Yes, I've noticed later when it ran in the CI. I've already updated it, although it looks there is an unrelated issue going on with Windows when running the tests, which also happens in the master branch #90

**Comment by fzyzcjy** [comment]
@davidmartos96 I see, that does look like a bug prior to this PR!

**Comment by fzyzcjy** [comment]
btw, do you have a windows machine? I only have mac currently, so it will not be very easy to debug what's going on in windows.

If we can somehow make a minimal reproducible sample, then we can create a bug at official Dart repository - since I guess this behavior of vm service can be a bug.

**Comment by davidmartos96** [comment]
@fzyzcjy I think I'm on the right track for the issue on master.
It works on Dart 2.19.6 and it fails on 3.0 and later.

When passing force=true to reloadSources it works as expected. When adding a small delay before reloading sources it works as expected.

image

 Then I found this: https://github.com/dart-lang/sdk/issues/51937
Could it be that there is an internal bug on Dart for Windows where the precision of the modified files is not correct? 

We could use force=true in the meantime, it's not like interactive should be used on big projects where hot reload performance should matter. What do you think?

**Comment by davidmartos96** [comment]
Or maybe it doesn't even matter for the `dart_interactive` case. Does isolate's sources include external packages? If not, force true would only be affecting a few dart files

**Comment by fzyzcjy** [comment]
Good job in finding the bug!

> Could it be that there is an internal bug on Dart for Windows where the precision of the modified files is not correct?

That sounds reasonable. A wild guess: can it be caused by, for example, precision is 1s instead of 1ms? (just wild guess, not searched anything)

> Or maybe it doesn't even matter for the dart_interactive case. Does isolate's sources include external packages? If not, force true would only be affecting a few dart files

I am not sure. Users can freely import anything, so it may include?

**Review by fzyzcjy — APPROVED** [review]
Reviewed and it looks great!

**Comment by davidmartos96** [comment]
@fzyzcjy  I've just filed a separate issue in the dart/sdk repo if you want to include it in the comments 
https://github.com/dart-lang/sdk/issues/53546

**Pull request merged**